feat: implement new Table component with enhanced features

limichange opened the pull request:
<img width="471" height="95" alt="image" src="https://github.com/user-attachments/assets/dabfcc8e-df5b-4f32-9047-817c87f705ce" />


- Introduced a new Table component that supports customizable rendering, sorting, and drag-and-drop functionality.
- Added Column and HeaderColumn components for better structure and sorting capabilities.
- Created utility functions for sorting logic and integrated them into the table's header interactions.
- Developed a TableGallery to showcase various table functionalities, including basic usage, custom rendering, and interactive features.
- Added types for table columns and props to improve type safety and developer experience.

This update enhances the data presentation capabilities within the application, providing a more flexible and user-friendly table component.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* New Features
  * Introduced sortable table headers with visual sort indicators and alignment options.
  * Enhanced column rendering: custom cell content, optional widths, and header/title styling.
  * Added interactive sorting via header clicks, including reset behavior.
  * Maintains skeleton loading rows and sticky header support.

* Documentation
  * Added Table Gallery with three demos: Basic, Custom Rendering, and Interactive Sorting.
  * Included usage tips, suggestions, and boundary conditions within the gallery.
  * New gallery route entry for the Table component.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

limichange committed:
feat: implement new Table component with enhanced features

- Introduced a new Table component that supports customizable rendering, sorting, and drag-and-drop functionality.
- Added Column and HeaderColumn components for better structure and sorting capabilities.
- Created utility functions for sorting logic and integrated them into the table's header interactions.
- Developed a TableGallery to showcase various table functionalities, including basic usage, custom rendering, and interactive features.
- Added types for table columns and props to improve type safety and developer experience.

This update enhances the data presentation capabilities within the application, providing a more flexible and user-friendly table component.

limichange committed:
Merge branch 'x' into OK-42924

revan-zhang commented:
### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/e662efac-39c0-4cac-8c55-c5c3ddb384a1?refs=4c80d21974feaf05327b602623eabf886bfbb6ae&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/e662efac-39c0-4cac-8c55-c5c3ddb384a1/license?refs=4c80d21974feaf05327b602623eabf886bfbb6ae&source=prChecksComment)

limichange committed:
fix: remove TableRowComponent prop from TableSkeleton

- Eliminated the TableRowComponent prop from the TableSkeleton component to streamline its implementation.
- This change simplifies the component structure and enhances maintainability.

limichange committed:
refactor: streamline TableSkeleton implementation and update sorting icon color

- Removed the TableSkeletonRow component to simplify the TableSkeleton structure.
- Updated the TableSkeleton import in the Table component to reflect this change.
- Adjusted the sorting icon color logic in useSortIcon for improved visual feedback based on active state.

This update enhances the maintainability of the table components and improves the user interface experience.

limichange commented:
@coderabbitai resolve

limichange committed:
Merge branch 'x' into OK-42924
